backend/business-partner-agent/src/main/java/org/hyperledger/bpa/model/ChatMessage.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.micronaut.data.annotation.AutoPopulated;
+import io.micronaut.data.annotation.DateCreated;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import java.time.Instant;
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@Table(name = "chat_message")
+public class ChatMessage {
+
+    @Id
+    @AutoPopulated
+    private UUID id;
+
+    @OneToOne
+    private Partner partner;
+
+    private String content;
+
+    private boolean incoming;
+
+    @JsonIgnore
+    @DateCreated
+    private Instant createdAt;
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    public Long timestamp() {
```

Comment:
Renaming the method getTimestamp has the same effect. To be consistent:

```java
@JsonIgnore
@DateCreated
@Column(name="created_at"
private Instant createdTs;

public Long getCreatedAt() {
    return this.createdTs.toEpochMilli();
}
```

Reply:
oh, nice!

Reply:
couldn't quite get that to work, think it gets confused because a column name matches a getter. tried a few different approaches to match that and didn't get it right. 🤷

Reply:
I see. Then I would just rename the timestamp method to

```java
public Long getCreatedAtTs() {
    return this.createdTs.toEpochMilli();
}
```
because the name timestamp is a bit ambiguous.

Reply:
👍